tests/spread/ci/task.yaml

```diff
@@ -0,0 +1,51 @@
+summary: Build an LXD spread test image with TPM software
+manual: true
+
+execute: |
+    INSTANCE_NAME=spread-test
+    IMAGE_NAME=lxd-spread-core20-img
+
+    # be in project folder
+    cd $PROJECT_PATH
+
+    # install prerequisites
+    snap install google-cloud-sdk --classic
+    snap install lxd --channel=latest/stable
+    lxd init --auto
+
+    # authenticate with gcloud
+    gcloud auth activate-service-account --key-file="./sa.json"
+
+    # launch the new lxc instance from an ubuntu 20.04 base
+    lxc launch ubuntu:20.04 $INSTANCE_NAME
+
+    # wait a few seconds before proceeding, otherwise we are going to have issues
+    # with the container (systemd) not being ready yet, and then the rest of
+    # the commands will fail
+    sleep 5s
+    lxc exec $INSTANCE_NAME -- bash -c 'systemctl is-system-running --wait' || true
+
+    # we build libtpms and swptm from source and preinstall that in the image for TPM emulation support
+    lxc exec $INSTANCE_NAME -- bash -c "apt update -yqq"
+    lxc exec $INSTANCE_NAME -- bash -c "apt install snapd ovmf qemu-system-x86 sshpass whois coreutils net-tools iproute2 automake autoconf libtool gcc build-essential libssl-dev dh-exec pkg-config dh-autoreconf libtasn1-6-dev libjson-glib-dev libgnutls28-dev expect gawk socat libseccomp-dev make -yqq"
+    lxc exec $INSTANCE_NAME -- bash -c "git clone https://github.com/stefanberger/libtpms"
+    lxc exec $INSTANCE_NAME -- bash -c "git clone https://github.com/stefanberger/swtpm"
+    lxc exec $INSTANCE_NAME -- bash -c "cd libtpms && ./autogen.sh --with-openssl --prefix=/usr --with-tpm2 && make -j4 && make check && make install"
+    lxc exec $INSTANCE_NAME -- bash -c "cd swtpm && ./autogen.sh --with-openssl --prefix=/usr && make -j4 && make -j4 check && make install"
+    lxc exec $INSTANCE_NAME -- bash -c "rm -rf libtpms"
+    lxc exec $INSTANCE_NAME -- bash -c "rm -rf swtpm"
+
+    # stop the container as the last step, the container is now ready for publishing
+    lxc stop $INSTANCE_NAME
+
+    # make a tarball out of the container image and delete the container
+    lxc publish $INSTANCE_NAME --alias ucspread
+    lxc image export ucspread ./$IMAGE_NAME
+
+    # upload the image file as https://storage.googleapis.com/snapd-spread-core/lxd/$IMAGE_NAME.tar.gz
+    gsutil -o GSUtil:parallel_composite_upload_threshold=2000M cp "./$IMAGE_NAME.tar.gz" "gs://snapd-spread-core/lxd/$IMAGE_NAME.tar.gz"
+
+restore: |
+    INSTANCE_NAME=spread-test
+    lxc stop $INSTANCE_NAME
```

Comment:
I would include || true because the command could failed in case the instance is not created 

lxc stop $INSTANCE_NAME || true
lxc delete $INSTANCE_NAME || true

Something important about spread is that in case a restore fails, then the machine is discarded, so all the tests that are executed after that one are aborted.

Reply:
perhaps you can stop the container during execute and then do in restore

lxc delete -f $INSTANCE_NAME || true

Reply:
That makes a lot of sense, I'll get that added